datafusion/core/src/execution/context.rs

```diff
@@ -2617,4 +2690,96 @@ mod tests {
                 .unwrap()
         }
     }
+
+    /// helper for the following drop schema tests
```

Comment:
I think we could use sqllogic tests https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/sqllogictests/ to write this more concisely

For example, perhaps we could extend:

https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/sqllogictests/test_files/ddl.slt